### ISSUE TITLE
Extra "SOL" character in player key in array

### ISSUE THREAD

**Issue by stephen304** [Austinb/GameQ#87]
In the results, if you try to do 
`foreach ($server['MC']['players'] as $player) {
        echo ($player['player']);
    }`

It won't work. If you look at the source of a print_r of the array and copy/paste the "player" (singular) key into notepad++, you will find an "SOL" character at the start of the key name. Leaving this character in notepad++ in my code allows the code to work. I'm not quite sure where this character comes from, but it might be good to fix. I haven't had any issues with any other keys in PHP.

**Comment by Austinb** [comment]
Can you please provide and example of what you are refering to?  The ip:port of the server will be fine I would like to see what EOL character you are seeing. You said SOL in title I assume it is EOL as in end of line like a \n, \r\n, etc...

Thanks

**Comment by Austinb** [comment]
Actually did you mean Start of line with an EOL character?  So some kind of tab (\t) or something in the actual result? Or is it just a weird character in there?

Still need need the server info to check against.

**Comment by stephen304** [comment]
Edit: SOH I meant SOH whoops.

Maybe SOH counts as an EOL character? It shows up as SOH in notepad++.
query [redacted] type minecraft
Get the array and try to extract the name of player 0 from the array. You will find that without this SOH (0x01) preceding the player name key "player", you will be unable to get the name of the player.

If you print_r the array and copy/paste the array key "player" and paste it into the code like so:
`$name = $server['MC']['players'][0]['SOLPlayer']` (copy/pasted not typed)

You will be able to successfully extract the name of the player.

Example: I did this:
`print_r($results['MC']['players'][0]);`
it returned:
`Array
(
    [player] => Stephen304
    [gq_name] => 
    [gq_kills] => 
    [gq_deaths] => 
    [gq_score] => 
    [gq_ping] => 
)`

So I did this:
`print_r($results['MC']['players'][0]['player']);`
and I get this:
`Notice: Undefined index: player`

Going a step further, I copy "player" from `[player] => Stephen304` (from my browser window) and I paste it into the second example. It shows up as:
`print_r($results['MC']['players'][0]['SOHplayer']);`
in notepad++. This returns:
`Stephen304`
as expected.

I suspect the code is inserting this extra control character somewhere which is mucking up php when you try to access the player names. I hope you can reproduce this error? I will stay on my server if you need to test.

If you don't get anything from my server, you can probably test any other minecraft server. I run unmodded and query defaults to the same port as the server itself (25565).

**Comment by Austinb** [comment]
I was able to find the issue.  It was a gamespy3 issue from where I changed the old code to a newish code because of a bug present in arma2 (#71 & commit: 68a9273).  This actually affected both players and team info (since minecraft doesnt have teams this didnt matter).

Basically the character issues were as follows:
The player designation from the return was actually showing up as \x01player_
The team designation from the return was actually showing up as \x00\x02team_t

Those invisible characters were being put into the actual key preventing the key reference of "player" from working (as you stated in your issue). 

Will push up a fix shortly.  Big thanks for finding this bug.
